### PR TITLE
Maven framework support

### PR DESCRIPTION
**Pull request by recreationalcode**
Added maven framework support
Split source and test classes according to common paradigm
Added a few lines to .gitignore to remove clutter from GitHub client
Refactored some Json code to use supported libraries

### CODE REVIEW

File: src/main/java/WeightedKing.java [hunk 1]
```diff
@@ -14,7 +14,7 @@ public WeightedKing(int i, int n, Value V, MsgHandler msg, ArrayList<Double> wei
 
     @Override
     public void runPhaseOne(){
-        double s0 = 0.0, s1 = 0.0, su = 0.0;
+        double s0 = 0.0, s1 = 0.0;
```
Comment (bethrichardson) [inline src/main/java/WeightedKing.java:17]: are we not supposed to be using the undecided weight?

Comment (recreationalcode) [inline src/main/java/WeightedKing.java:17]: Nah, not in this phase. InteliiJ told me it was unused.

**Comment by recreationalcode** [comment]
Thanks Beth!

**Comment by recreationalcode** [comment]
Just for reference:
- Open the project by opening the Maven POM file itself.
- Make sure the src/main/java folder and src/test/java folders are your source and test directories respectively
- Make sure the target folder is an excluded directory
- Set the Project SDK to Java 1.7
- Set the Project and Module language level to 7.0
- Allow import of Maven dependencies (IntelliJ should prompt you)
- You should be good to go!

**Pull request merged**